### PR TITLE
Pull request #1

### PR DESCRIPTION
**Pull request by doztrock**
I've added the `grub.cfg` needed to generate an ISO image, now it runs with:

`qemu-system-i386 -cdrom exylus.iso`

Also, I've left the 'run' default-option in the Makefile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,5 +35,12 @@ kernel: $(KERNEL_CC_LIST) $(KERNEL_AS_LIST)
...
 
+iso: kernel
+	cp exylus.bin iso/boot/exylus.bin
+	grub-mkrescue -o exylus.iso iso/
+
+run-iso: iso
+	qemu-system-i386 -cdrom exylus.iso
+
 clean:
 	find -type f \( -name "*.o" -o -name "*.bin" \) -delete
```
Comment (ketanhwr) [inline Makefile:47]: You should add a filter for `.iso` files here. Add a newline at the end of the file, I missed it somehow.

File: iso/boot/grub/grub.cfg [hunk 2]
```diff
@@ -0,0 +1,7 @@
+set timeout=1
+set default=0
```
Comment (ketanhwr) [inline iso/boot/grub/grub.cfg:2]: I haven't used `timeout` and `default` variables. Why are they required?

File: kernel/cpu/idt.c [hunk 3]
```diff
@@ -63,7 +63,7 @@ static void init_idt_table()
 	idt_flush((uint32_t)&idt_ptr);
 }
 
-void idt_set_gate(uint8_t num, uint32_t base, uint16_t sel, uint8_t flags)
+void idt_set_gate(int num, uint32_t base, uint16_t sel, uint8_t flags)
```
Comment (ketanhwr) [inline kernel/cpu/idt.c:66]: Your pull request has nothing to do with this, please revert this change.

File: kernel/include/idt.h [hunk 4]
```diff
@@ -58,4 +58,4 @@ extern void isr31();
 
 void init_idt();
 
-void idt_set_gate(uint8_t num, uint32_t base, uint16_t sel, uint8_t flags);
+void idt_set_gate(int num, uint32_t base, uint16_t sel, uint8_t flags);
```
Comment (ketanhwr) [inline kernel/include/idt.h:61]: This too.

**Review by ketanhwr — CHANGES_REQUESTED** [review]
Thanks for your pull request! I've made some inline comments, please see to them.

There's no requirement of the three include files you've added, you can remove them (why did you add them in the first place?). Also from next time, the indentation is supposed to be a tab of size 4 (without spaces).

Thanks! :slightly_smiling_face:

File: iso/boot/grub/grub.cfg (hunk 2 above)
Comment (doztrock) [reply iso/boot/grub/grub.cfg:2]: `timeout` -> This is a countdown before boot the operating system, in that case, the GRUB menu will be shown for a second before boot up the OS.

`default` -> The OS to be booted up by default, in that case, it will be the entry 0, because in the grub,cfg we can have many entries as possible, so the default entry will be 0 -> First

File: kernel/cpu/idt.c (hunk 3 above)
Comment (doztrock) [reply kernel/cpu/idt.c:66]: In the `stdint.h` standard file, there's a declaration (in my case at line 48).

`typedef unsigned char           uint8_t;`

So the type `uint8_t` is an `unsigned char`, if we try to compile, the compiler will show this message:

> kernel/cpu/idt.c:71:2: warning: array subscript has type 'char' [-Wchar-subscripts]
>   idt_entries[num].always0 = 0;

In that case i've set the `idt` index, like an int, because it's an `integer`, not a `char`, I've used it and the compilation works.

File: kernel/include/idt.h (hunk 4 above)
Comment (doztrock) [reply kernel/include/idt.h:61]: In the `stdint.h` standard file, there's a declaration (in my case at line 48).

`typedef unsigned char           uint8_t;`

So the type `uint8_t` is an `unsigned char`, if we try to compile, the compiler will show this message:

> kernel/cpu/idt.c:71:2: warning: array subscript has type 'char' [-Wchar-subscripts]
>   idt_entries[num].always0 = 0;

In that case i've set the `idt` index, like an int, because it's an `integer`, not a `char`, I've used it and the compilation works.

File: kernel/include/stdbool.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/*
```
Comment (doztrock) [inline kernel/include/stdbool.h:1]: I've added this file to handle the basic definition of `boolean` types, if we have an independent OS, it has to have all headers and declarations using a non-standard libc.

File: kernel/include/stddef.h [hunk 6]
```diff
@@ -0,0 +1,25 @@
+/*
```
Comment (doztrock) [inline kernel/include/stddef.h:1]: I've added this file to handle the basic definition of `NULL` and `size_t` types, if we have an independent OS, it has to have all headers and declarations using a non-standard libc.

File: kernel/include/stdint.h [hunk 7]
```diff
@@ -0,0 +1,65 @@
+/*
```
Comment (doztrock) [inline kernel/include/stdint.h:1]: I've added this file to handle the basic definition of `integer` types, I've base my file in the standard `stdint.h`, if we have an independent OS, it has to have all headers and declarations using a non-standard libc.

**Comment by doztrock** [comment]
I've made some comment about the reason why I've added these files. Hope it helps

**Comment by ketanhwr** [comment]
@doztrock we can add these files later on when I start working on a libc. Currently you can remove them since they are not required at the present moment. I think you forgot making the changes in the makefile I mentioned.

**Comment by ketanhwr** [comment]
@doztrock what happened?

**Comment by doztrock** [comment]
I'm not an expert with Github, sorry. I was trying to undo my changes, and I've tought it was closing the pull request, sorry haha

**Comment by ketanhwr** [comment]
Alright. Waiting to merge this soon, please make the changes fast!

**Comment by doztrock** [comment]
I close this pull request, i will open a new one, to be more organizated with the commits :)

**Comment by ketanhwr** [comment]
@doztrock it doesn't matter, I'll squash the commits into one commit and then merge so it won't be a problem if the commits are not organised.

**Comment by doztrock** [comment]
@ketanhwr Well, I prefer some organizated commits :), anyway I've created the new pull request, please you can review.
